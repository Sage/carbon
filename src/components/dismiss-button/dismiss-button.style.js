import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';
import BaseTheme from '../../style/themes/base';
import OptionsHelper from '../../utils/helpers/options-helper';
import CloseIconClassicStyling from './dismiss-button-classic.style';
import Link from '../link';
import StyledIcon from '../icon/icon.style';
import { isClassic } from '../../utils/helpers/style-helper';

const DismissButtonStyle = styled.div`
  border: none;
  position: absolute;
  right: 16px;
  ${({ theme }) => !isClassic(theme) && css`
    margin-top: -10px;
    top: 50%;
  `}
  ${StyledIcon} {
    &:before {
      color: ${({ theme }) => theme.colors.border};
    }

    &:hover:before{
      color: ${({ theme }) => theme.colors.focusedIcon};
    }
  }

  ${CloseIconClassicStyling}
`;

const LinkStyle = styled(Link)`
  ${({ theme }) => !isClassic(theme) && css`
    a:focus {
      outline: none;
      background-color: transparent;
       span {
        &:before {
          outline: 2px solid ${theme.colors.focus};
          outline-offset: 3px;
        }
      }
    }
`}

  .carbon-link__content {
    display: none;
  }

  ${StyledIcon} {
    margin-right: 0;
    top: -2px;
  }
`;

DismissButtonStyle.defaultProps = {
  variant: 'info',
  roundedCorners: true,
  theme: BaseTheme,
  transparent: false
};

DismissButtonStyle.propTypes = {
  variant: PropTypes.oneOf(OptionsHelper.colors),
  border: PropTypes.bool,
  roundedCorners: PropTypes.bool,
  transparent: PropTypes.bool
};

LinkStyle.defaultProps = {
  theme: BaseTheme
};

export { DismissButtonStyle, LinkStyle };
