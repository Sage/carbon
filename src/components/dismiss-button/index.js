export { default } from './dismiss-button.component';
