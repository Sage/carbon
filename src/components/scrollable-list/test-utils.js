import React from 'react';
import { makeArrayKeys } from '../../__spec_helper__/test-utils';
import ScrollableListItem from './scrollable-list-item.component';

const listItemReducer = ({ nonSelectables = [], customSelectables = [] }) => {
  // generate jsx for selectable list items, based on indexes in config
  return (acc, item, index) => {
    if (nonSelectables.includes(item)) return [...acc, <div />];
    if (customSelectables.includes(item)) return [...acc, <div isSelectable />];

    return [...acc, <ScrollableListItem id={ index } />];
  };
};

const renderListItems = (opts) => {
  return makeArrayKeys(opts.num).reduce(listItemReducer(opts), []);
};

export default renderListItems;
