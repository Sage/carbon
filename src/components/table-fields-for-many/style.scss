@import "colors";

.ui-table-fields-for-many {
  border: 1px solid $border-color {
    collapse: separate;
    radius: 4px;
    spacing: 0;
  }
}
