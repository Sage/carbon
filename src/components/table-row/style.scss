@import "colors";

$row__background-color--even: $white;
$row__background-color--odd: #F2F2F2;
$row__background-color--hover: #D2EED1;
$row__border-radius: 4px;
$row__delete-icon-size: 18px;
$row__delete-icon-color--normal: #AFAFAF;
$row__delete-icon-color--hover: $error;

.ui-table-row {
  background-color: $row__background-color--odd;

  // first row
  &:first-child {
    .ui-table-row__td {
      // first td in first row
      &:first-child {
        border-radius: $row__border-radius 0 0 0;
      }

      // last td in first row
      &:last-child {
        border-radius: 0 $row__border-radius 0 0;
      }
    }
  }

  // last row
  &:last-child {
    .ui-table-row__td {
      // first td in last row
      &:first-child {
        border-radius: 0 0 0 $row__border-radius;
      }

      // last td in last row
      &:last-child {
        border-radius: 0 0 $row__border-radius 0;
      }
    }
  }

  // even rows
  &:nth-child(even) {
    background-color: $row__background-color--even;
  }

  // row hover
  &:hover {
    background-color: $row__background-color--hover;

    // delete icon on row hover
    .ui-table-row__delete-icon:before {
      content: "w";
      font-size: $row__delete-icon-size;
    }
  }
}

.ui-table-row__td {
  padding: 4px 2px;
}

.ui-table-row__delete {
  background: none;
  border: none;
  cursor: pointer;
  display: block;
  padding: 0;
  width: $row__delete-icon-size;
}

.ui-table-row__delete-icon {
  &:before {
    color: $row__delete-icon-color--normal;
    font-size: $row__delete-icon-size - 2;
  }

  &:hover:before {
    color: $row__delete-icon-color--hover;
  }
}
