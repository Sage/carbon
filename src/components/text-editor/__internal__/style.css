
.textBold {
  font-weight: bold;
}
.textItalic {
  font-style: italic;
}
