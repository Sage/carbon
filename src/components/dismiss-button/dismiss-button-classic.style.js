import { css } from 'styled-components';
import { THEMES } from '../../style/themes';

const colors = {
  error: '#c7384f',
  warning: '#ff7d00',
  success: '#50b848',
  info: '#1573e6',
  default: '#335c6d',
  help: '#ffab00',
  maintenance: '#ff7d00',
  new: '#639'
};

export default ({ theme, variant }) => theme.name === THEMES.classic && css`
  align-items: center;
  display: flex;
  margin-left: auto;
  margin-right: 15px;
  justify-content: center;
  text-align: center;
  position: static;
  min-width: 15px;
  
  .icon-close {
    &:before {
      color: ${colors[variant]};
      font-size:16px;
    }
    &:hover:before{
      color: ${colors[variant]};
    }
  }

  ${({ transparent }) => transparent && css`
    background-color: transparent;
  `}
`;
