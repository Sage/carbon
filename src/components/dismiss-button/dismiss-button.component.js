import React from 'react';
import PropTypes from 'prop-types';
import { DismissButtonStyle, LinkStyle } from './dismiss-button.style';
import OptionsHelper from '../../utils/helpers/options-helper';

const DismissButton = ({ variant, onDismiss, transparent }) => {
  return (
    <DismissButtonStyle variant={ variant } transparent={ transparent }>
      <LinkStyle
        icon='close' data-element='dismiss'
        onClick={ onDismiss }
      />
    </DismissButtonStyle>
  );
};

DismissButton.defaultProps = {
  variant: 'info',
  transparent: false
};

DismissButton.propTypes = {
  variant: PropTypes.oneOf(OptionsHelper.colors),
  onDismiss: PropTypes.func,
  transparent: PropTypes.bool
};

export default DismissButton;
