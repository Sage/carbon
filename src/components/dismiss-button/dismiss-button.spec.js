import React from 'react';
import TestRenderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import DismissButton from './dismiss-button.component';
import { DismissButtonStyle, LinkStyle } from './dismiss-button.style';
import 'jest-styled-components';
import OptionsHelper from '../../utils/helpers/options-helper/options-helper';
import classicTheme from '../../style/themes/classic';
import { assertStyleMatch } from '../../__spec_helper__/test-utils';

function render(props) {
  return TestRenderer.create(<DismissButtonStyle { ...props } />);
}

describe('DismissButton', () => {
  describe('when rendered', () => {
    it('should render correctly', () => {
      expect(shallow(<DismissButton />)).toHaveLength(1);
    });
  });

  describe('when render with no additional props', () => {
    OptionsHelper.messages.forEach((variant) => {
      it(`should match snapshot for ${variant}`, () => {
        const wrapper = render({ variant });
        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});

describe('when in classic mode', () => {
  OptionsHelper.colors.forEach((variant) => {
    describe(`when rendered as ${variant}`, () => {
      it('should match the snapshot', () => {
        const wrapper = render({ variant, theme: classicTheme });
        expect(wrapper).toMatchSnapshot();
      });

      describe('when transparent prop is set to true', () => {
        it('should apply transparent background', () => {
          const wrapper = render({
            transparent: true,
            theme: classicTheme,
            variant
          });
          expect(wrapper).toMatchSnapshot();
        });
      });
    });
  });
});

describe('LinkStyle', () => {
  it('applies peoper styling when focused', () => {
    const wrapper = TestRenderer.create(<LinkStyle />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('DismissButtonStyle', () => {
  it('should render correct style based on classic theme', () => {
    assertStyleMatch({
      backgroundColor: 'transparent'
    }, mount(<DismissButtonStyle transparent theme={ classicTheme } />));
  });
});
