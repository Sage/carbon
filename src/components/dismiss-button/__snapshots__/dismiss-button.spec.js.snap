// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DismissButton when render with no additional props should match snapshot for error 1`] = `
.c0 {
  border: none;
  position: absolute;
  right: 16px;
  margin-top: -10px;
  top: 50%;
}

.c0 .c1:before {
  color: #668491;
}

.c0 .c1:hover:before {
  color: #335B6D;
}

<div
  className="c0"
/>
`;

exports[`DismissButton when render with no additional props should match snapshot for info 1`] = `
.c0 {
  border: none;
  position: absolute;
  right: 16px;
  margin-top: -10px;
  top: 50%;
}

.c0 .c1:before {
  color: #668491;
}

.c0 .c1:hover:before {
  color: #335B6D;
}

<div
  className="c0"
/>
`;

exports[`DismissButton when render with no additional props should match snapshot for success 1`] = `
.c0 {
  border: none;
  position: absolute;
  right: 16px;
  margin-top: -10px;
  top: 50%;
}

.c0 .c1:before {
  color: #668491;
}

.c0 .c1:hover:before {
  color: #335B6D;
}

<div
  className="c0"
/>
`;

exports[`DismissButton when render with no additional props should match snapshot for warning 1`] = `
.c0 {
  border: none;
  position: absolute;
  right: 16px;
  margin-top: -10px;
  top: 50%;
}

.c0 .c1:before {
  color: #668491;
}

.c0 .c1:hover:before {
  color: #335B6D;
}

<div
  className="c0"
/>
`;

exports[`LinkStyle applies peoper styling when focused 1`] = `
.c1 {
  display: inline-block;
}

.c1 a {
  font-size: 14px;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #26A826;
  display: inline-block;
}

.c1 a .c2 {
  margin-right: 5px;
  position: relative;
  vertical-align: middle;
}

.c1 a:hover {
  cursor: pointer;
  color: #006300;
}

.c1 a:focus {
  color: rgba(0,0,0,0.9);
  cursor: default;
  background-color: #FFDA7F;
  outline: none;
}

.c0 a:focus {
  outline: none;
  background-color: transparent;
}

.c0 a:focus span:before {
  outline: 2px solid #FFB500;
  outline-offset: 3px;
}

.c0 .carbon-link__content {
  display: none;
}

.c0 .c2 {
  margin-right: 0;
  top: -2px;
}

<div
  className="c0 c1"
  data-component="link"
  disabled={false}
>
  <a
    onClick={[Function]}
    onKeyDown={[Function]}
    tabIndex="0"
  >
    <span
      className="carbon-link__content"
    />
  </a>
</div>
`;

exports[`when in classic mode when rendered as default should match the snapshot 1`] = `
.c0 {
  border: none;
  position: absolute;
  right: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: auto;
  margin-right: 15px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  position: static;
  min-width: 15px;
}

.c0 .c1:before {
  color: #668491;
}

.c0 .c1:hover:before {
  color: #335B6D;
}

.c0 .c1:before {
  color: #335c6d;
  font-size: 16px;
}

.c0 .c1:hover:before {
  color: #335c6d;
}

<div
  className="c0"
/>
`;

exports[`when in classic mode when rendered as default when transparent prop is set to true should apply transparent background 1`] = `
.c0 {
  border: none;
  position: absolute;
  right: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: auto;
  margin-right: 15px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  position: static;
  min-width: 15px;
  background-color: transparent;
}

.c0 .c1:before {
  color: #668491;
}

.c0 .c1:hover:before {
  color: #335B6D;
}

.c0 .c1:before {
  color: #335c6d;
  font-size: 16px;
}

.c0 .c1:hover:before {
  color: #335c6d;
}

<div
  className="c0"
/>
`;

exports[`when in classic mode when rendered as error should match the snapshot 1`] = `
.c0 {
  border: none;
  position: absolute;
  right: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: auto;
  margin-right: 15px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  position: static;
  min-width: 15px;
}

.c0 .c1:before {
  color: #668491;
}

.c0 .c1:hover:before {
  color: #335B6D;
}

.c0 .c1:before {
  color: #c7384f;
  font-size: 16px;
}

.c0 .c1:hover:before {
  color: #c7384f;
}

<div
  className="c0"
/>
`;

exports[`when in classic mode when rendered as error when transparent prop is set to true should apply transparent background 1`] = `
.c0 {
  border: none;
  position: absolute;
  right: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: auto;
  margin-right: 15px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  position: static;
  min-width: 15px;
  background-color: transparent;
}

.c0 .c1:before {
  color: #668491;
}

.c0 .c1:hover:before {
  color: #335B6D;
}

.c0 .c1:before {
  color: #c7384f;
  font-size: 16px;
}

.c0 .c1:hover:before {
  color: #c7384f;
}

<div
  className="c0"
/>
`;

exports[`when in classic mode when rendered as help should match the snapshot 1`] = `
.c0 {
  border: none;
  position: absolute;
  right: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: auto;
  margin-right: 15px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  position: static;
  min-width: 15px;
}

.c0 .c1:before {
  color: #668491;
}

.c0 .c1:hover:before {
  color: #335B6D;
}

.c0 .c1:before {
  color: #ffab00;
  font-size: 16px;
}

.c0 .c1:hover:before {
  color: #ffab00;
}

<div
  className="c0"
/>
`;

exports[`when in classic mode when rendered as help when transparent prop is set to true should apply transparent background 1`] = `
.c0 {
  border: none;
  position: absolute;
  right: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: auto;
  margin-right: 15px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  position: static;
  min-width: 15px;
  background-color: transparent;
}

.c0 .c1:before {
  color: #668491;
}

.c0 .c1:hover:before {
  color: #335B6D;
}

.c0 .c1:before {
  color: #ffab00;
  font-size: 16px;
}

.c0 .c1:hover:before {
  color: #ffab00;
}

<div
  className="c0"
/>
`;

exports[`when in classic mode when rendered as info should match the snapshot 1`] = `
.c0 {
  border: none;
  position: absolute;
  right: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: auto;
  margin-right: 15px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  position: static;
  min-width: 15px;
}

.c0 .c1:before {
  color: #668491;
}

.c0 .c1:hover:before {
  color: #335B6D;
}

.c0 .c1:before {
  color: #1573e6;
  font-size: 16px;
}

.c0 .c1:hover:before {
  color: #1573e6;
}

<div
  className="c0"
/>
`;

exports[`when in classic mode when rendered as info when transparent prop is set to true should apply transparent background 1`] = `
.c0 {
  border: none;
  position: absolute;
  right: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: auto;
  margin-right: 15px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  position: static;
  min-width: 15px;
  background-color: transparent;
}

.c0 .c1:before {
  color: #668491;
}

.c0 .c1:hover:before {
  color: #335B6D;
}

.c0 .c1:before {
  color: #1573e6;
  font-size: 16px;
}

.c0 .c1:hover:before {
  color: #1573e6;
}

<div
  className="c0"
/>
`;

exports[`when in classic mode when rendered as maintenance should match the snapshot 1`] = `
.c0 {
  border: none;
  position: absolute;
  right: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: auto;
  margin-right: 15px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  position: static;
  min-width: 15px;
}

.c0 .c1:before {
  color: #668491;
}

.c0 .c1:hover:before {
  color: #335B6D;
}

.c0 .c1:before {
  color: #ff7d00;
  font-size: 16px;
}

.c0 .c1:hover:before {
  color: #ff7d00;
}

<div
  className="c0"
/>
`;

exports[`when in classic mode when rendered as maintenance when transparent prop is set to true should apply transparent background 1`] = `
.c0 {
  border: none;
  position: absolute;
  right: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: auto;
  margin-right: 15px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  position: static;
  min-width: 15px;
  background-color: transparent;
}

.c0 .c1:before {
  color: #668491;
}

.c0 .c1:hover:before {
  color: #335B6D;
}

.c0 .c1:before {
  color: #ff7d00;
  font-size: 16px;
}

.c0 .c1:hover:before {
  color: #ff7d00;
}

<div
  className="c0"
/>
`;

exports[`when in classic mode when rendered as new should match the snapshot 1`] = `
.c0 {
  border: none;
  position: absolute;
  right: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: auto;
  margin-right: 15px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  position: static;
  min-width: 15px;
}

.c0 .c1:before {
  color: #668491;
}

.c0 .c1:hover:before {
  color: #335B6D;
}

.c0 .c1:before {
  color: #639;
  font-size: 16px;
}

.c0 .c1:hover:before {
  color: #639;
}

<div
  className="c0"
/>
`;

exports[`when in classic mode when rendered as new when transparent prop is set to true should apply transparent background 1`] = `
.c0 {
  border: none;
  position: absolute;
  right: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: auto;
  margin-right: 15px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  position: static;
  min-width: 15px;
  background-color: transparent;
}

.c0 .c1:before {
  color: #668491;
}

.c0 .c1:hover:before {
  color: #335B6D;
}

.c0 .c1:before {
  color: #639;
  font-size: 16px;
}

.c0 .c1:hover:before {
  color: #639;
}

<div
  className="c0"
/>
`;

exports[`when in classic mode when rendered as success should match the snapshot 1`] = `
.c0 {
  border: none;
  position: absolute;
  right: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: auto;
  margin-right: 15px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  position: static;
  min-width: 15px;
}

.c0 .c1:before {
  color: #668491;
}

.c0 .c1:hover:before {
  color: #335B6D;
}

.c0 .c1:before {
  color: #50b848;
  font-size: 16px;
}

.c0 .c1:hover:before {
  color: #50b848;
}

<div
  className="c0"
/>
`;

exports[`when in classic mode when rendered as success when transparent prop is set to true should apply transparent background 1`] = `
.c0 {
  border: none;
  position: absolute;
  right: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: auto;
  margin-right: 15px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  position: static;
  min-width: 15px;
  background-color: transparent;
}

.c0 .c1:before {
  color: #668491;
}

.c0 .c1:hover:before {
  color: #335B6D;
}

.c0 .c1:before {
  color: #50b848;
  font-size: 16px;
}

.c0 .c1:hover:before {
  color: #50b848;
}

<div
  className="c0"
/>
`;

exports[`when in classic mode when rendered as warning should match the snapshot 1`] = `
.c0 {
  border: none;
  position: absolute;
  right: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: auto;
  margin-right: 15px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  position: static;
  min-width: 15px;
}

.c0 .c1:before {
  color: #668491;
}

.c0 .c1:hover:before {
  color: #335B6D;
}

.c0 .c1:before {
  color: #ff7d00;
  font-size: 16px;
}

.c0 .c1:hover:before {
  color: #ff7d00;
}

<div
  className="c0"
/>
`;

exports[`when in classic mode when rendered as warning when transparent prop is set to true should apply transparent background 1`] = `
.c0 {
  border: none;
  position: absolute;
  right: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: auto;
  margin-right: 15px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  position: static;
  min-width: 15px;
  background-color: transparent;
}

.c0 .c1:before {
  color: #668491;
}

.c0 .c1:hover:before {
  color: #335B6D;
}

.c0 .c1:before {
  color: #ff7d00;
  font-size: 16px;
}

.c0 .c1:hover:before {
  color: #ff7d00;
}

<div
  className="c0"
/>
`;
